tests/test_reusable_executor.py

```diff
@@ -486,6 +488,46 @@ def test_call_item_gc_crash_or_exit(self, bad_object, match):
             for _ in executor.map(sleep, [.1] * 100):
                 pass
 
+    def test_sigkill_shutdown_leaks_workers(self):
+        # Create a parent process which will report its workers pids
+        # and sigkill itself.
+        code = """if True:
+        import os
+
+        import loky
+        import psutil
+
+        parent_pid = os.getpid()
+        with loky.get_reusable_executor(timeout=1, kill_workers=True) as p:
+            list(p.map(lambda x: x, range(100)))
+            for pid in p._processes.keys():
+                print(f'worker_pid:{pid}')
+            print(f'parent_pid:{parent_pid}')
+            psutil.Process(os.getpid()).kill()
+        """
+        with NamedTemporaryFile(mode='w', suffix="_joblib.py",
+                                delete=True) as f:
+            f.write(code)
+            f.flush()
+            cmd = [sys.executable, f.name]
+            out = subprocess.run(
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+                text=True,
+            )
```

Comment:
The test hangs here if the workers do not exit I guess:
```suggestion
            out = subprocess.run(
                cmd,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                text=True,
                timeout=60
            )
```

Reply:
:muscle:  
So are we're good with the test's timeout failure (in case the issue isn't fixed ) ?

Reply:
Yes, I will merge once this is green.